File: python/14.py
class Solution(object):
    def every_possible_sequence(self,word):
        res=set()
        for len1 in range(1,len(word)):
            for start in range(0,len(word)):
                res.add(word[start:start+len1])
        res.add(word)
        
        return res
            
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        זה הפתרון היפה שלי שבודק על כל רצף שקיים ולא רק בהתחלה
        """
        res=''
        temparr=[]
        for i in strs:
            temparr.append(len(i))
        min_len=min(temparr)
        arr=self.every_possible_sequence(strs[0])
        for i in arr:
            all_=True
            for j in strs:
                if i not in j:
                    all_=False
            if all_ and len(i)>len(res):
                res=i
        return res
            

File: javascript/14.js
// Write a function to find the longest common prefix string amongst an array of strings.

// If there is no common prefix, return an empty string "".

 

// Example 1:

// Input: strs = ["flower","flow","flight"]
// Output: "fl"
// Example 2:

// Input: strs = ["dog","racecar","car"]
// Output: ""
// Explanation: There is no common prefix among the input strings.
 

// Constraints:

// 1 <= strs.length <= 200
// 0 <= strs[i].length <= 200
// strs[i] consists of only lower-case English letters.


/**
 * @param {string[]} strs
 * @return {string}
 */
const longestCommonPrefix = function (strs) {
  let min = Number.MAX_VALUE;
  let short = "";
  for (let i = 0; i < strs.length; i++) {
    if (strs[i].length < min) {
      min = strs[i].length;
      short = strs[i];
    }
  }
  console.log(min, short);
  let shortest = all_prefix(short);
  let res = "";
  console.log(shortest);
  shortest.forEach((perfix) => {});
};

const all_prefix = function (str) {
  let arr = new Set();
  for (let i = 1; i < str.length; i++) {
    for (let j = 0; j + i <= str.length; j++) {
      // console.log(str.slice(j,j+i));
      // arr.push(str.slice(j,j+i));
      arr.add(str.slice(j, j + i));
    }
  }
  arr.add(str);
  console.log(arr);
  return arr;
};

console.log(longestCommonPrefix(["matan", "raz", "romi", "my"]));
